### ISSUE TITLE
mnist_siamese.py Example

### ISSUE THREAD

**Issue by JJN123** [keras-team/keras#9575]
In the example mnist_siamese.py:

https://github.com/keras-team/keras/blob/master/examples/mnist_siamese.py

Euclidean distance is defined as:

`def euclidean_distance(vects):
       x, y = vects
       return K.sqrt(K.maximum(K.sum(K.square(x - y), axis=1, keepdims=True), K.epsilon()))`

What is the purpose of the` K.Maximum` here?

**Comment by ozabluda** [comment]
The example reproduces the paper referenced in the example. It's explained there - so called "contrastive loss".

**Comment by JJN123** [comment]
The contrastive loss has a K.maximum, as explained in the paper, but I do not see why the Euclidean  distance has a K.maximum. My question is referring to the implementation of  euclidean_distance.

**Comment by ozabluda** [comment]
K.maximum in the euclidean_distance is to make sure the value under the square root is not too close to zero, otherwise the backprop derivative may become infinite (unstable).

**Comment by JJN123** [comment]
Right, makes sense. Thanks for the help.
